x-pack/plugins/observability/public/components/shared/exploratory_view/series_editor/columns/series_actions.tsx

```diff
@@ -44,8 +54,31 @@ export function SeriesActions({ seriesId, series, seriesConfig, onEditClick }: P
     } else {
       setSeries(seriesId, { ...series, hidden: true });
     }
+    closePopover();
   };
 
+  const onRemoveSeriesClick = () => {
```

Comment:
I would wrap this function in a `useCallback`

Reply:
Could you elaborate on why would you do that? I don't see the need in this case (in any of your comments)

Reply:
Sure, sorry :D

tl;dr `useCallback` will prevent unnecessary re-renders on the components that use these callbacks.

---

The component `SeriesActions` will eventually render multiple times over the lifetime of the page. On each render the browser creates and allocates an anonymous function for `onRemoveSeriesClick`, one for `changePopoverVisibility` and one for `closePopover`.

Those callback functions are now different on each rerender. (I tried to illustrate it here https://codesandbox.io/s/sharp-lehmann-6c3qp?file=/src/App.js). When using those functions in some of the components (for example the popover button) React will force a rerender of those components as well, even when functionally nothing has changed.

`useCallback` makes sure that the function is _exactly_ the same object on each call. React then can see that when comparing the components that use the functions and not touch them.

In general it's a good practice to wrap any function that is used as a callback (`onClick`, `onTouch`, etc) in `useCallback` to prevent these kind of rerenders.

Reply:
If I'm being honest... I'm not a big fan of _any_ hooks myself :D